promql/generated_parser.y

```diff
@@ -151,5 +161,30 @@ match_op        :
                         { yylex.(*parser).errorf("expected label matching operator but got %s", yylex.(*parser).token.val) } 
                 ;
 
+label_set       :
+                LEFT_BRACE label_set_list RIGHT_BRACE
+                        { $$ = labels.New($2...) }
+                | LEFT_BRACE RIGHT_BRACE
+                        { $$ = labels.New() }
+                ;
+
+label_set_list  :
+                label_set_list COMMA label_set_item
```

Comment:
Hmm, do we currently allow trailing commas?

Reply:
No.

What corresponds to `label_set_list` was previously parsed by the `labelMatchers()` function that is removed in this PR and is the same one that is used for label matchers in queries.

Reply:
Testing it we do, so this needs fixing and a unittest.

Reply:
Uh, I see.

This means #6410 did break the old behavior, too.

Reply:
We were bound to have some breakage through this process, at least it didn't make it into a release.

Reply:
The grammar *should* be fixed now.

I've added another unit test for the label matchers grammar from PR #6410 .

The labels set grammar isn't directly unit tested at all.

This is not as bad as it sounds, since it is only used by the `ParseMetric()` which is only used internally by unit tests. Still such tests should be added, possibly in another PR.

Reply:
Label matchers are now properly unit tested.

I'd consider adding unit tests for triggering error messages inside label sets less important, since label sets are only parsed by internal test routines anyway, and their parsing code looks almost the same as the one for label matchers.

If such tests are desired I can do it anyway.